src/index.js

```diff
@@ -25,7 +26,11 @@ Walker.prototype._read = function () {
   self.fs.lstat(pathItem, function (err, stats) {
     var item = { path: pathItem, stats: stats }
     if (err) return self.emit('error', err, item)
-    if (!stats.isDirectory()) return self.push(item)
+
+    if (!stats.isDirectory() || (self.rootDepth &&
+      pathItem.split(path.sep).length - (self.rootDepth + 1) >= self.options.depthLimit)) {
```

Comment:
`pathItem.split(path.sep)`: how does this handle paths like `foo//bar\baz` & `C:\\Users\donald_duck`? Should we be `path.normalize()`ing here?